subprojects/core/src/main/java/org/gradle/api/internal/tasks/options/FieldOptionElement.java

```diff
@@ -129,6 +135,12 @@ public PropertyFieldSetter(Method getter, Field field) {
             this.elementType = ModelType.of(getter.getGenericReturnType()).getTypeVariables().get(0).getRawClass();
         }
 
+        public PropertyFieldSetter(Method getter, Field field, Class<?> elementType) {
+            this.getter = getter;
+            this.field = field;
+            this.elementType = elementType;
```

Comment:
🧐 The other constructor should call this one. It's best to have all constructors converge on one to reduce code duplication (DRY).